diff -ur a/package/feeds/luci/luci-app-ssr-plus/root/usr/bin/ssr-rules b/package/feeds/luci/luci-app-ssr-plus/root/usr/bin/ssr-rules
--- a/package/feeds/luci/luci-app-ssr-plus/root/usr/bin/ssr-rules	2023-09-12 09:17:19.289970874 +0800
+++ b/package/feeds/luci/luci-app-ssr-plus/root/usr/bin/ssr-rules	2023-09-12 09:18:58.473302323 +0800
@@ -253,15 +253,15 @@
 	router)
 		$ipt -A SS_SPEC_TPROXY -p udp -m set --match-set ss_spec_wan_ac dst -j RETURN
 		$ipt -A SS_SPEC_TPROXY -p udp -m set --match-set china dst -j RETURN
-		$ipt -A SS_SPEC_TPROXY -p udp --dport 80 -j REJECT
-		$ipt -A SS_SPEC_TPROXY -p udp --dport 443 -j REJECT
+		$ipt -A SS_SPEC_TPROXY -p udp --dport 80 -j DROP
+		$ipt -A SS_SPEC_TPROXY -p udp --dport 443 -j DROP
 		$ipt -A SS_SPEC_TPROXY -p udp -m set --match-set gmlan src -m set ! --match-set china dst -j TPROXY --on-port "$LOCAL_PORT" --tproxy-mark 0x01/0x01
 		$ipt -A SS_SPEC_TPROXY -p udp $PROXY_PORTS -m set ! --match-set ss_spec_wan_ac dst -j TPROXY --on-port "$LOCAL_PORT" --tproxy-mark 0x01/0x01
 		;;
 	gfw)
 		$ipt -A SS_SPEC_TPROXY -p udp -m set --match-set china dst -j RETURN
-		$ipt -A SS_SPEC_TPROXY -p udp --dport 80 -j REJECT
-		$ipt -A SS_SPEC_TPROXY -p udp --dport 443 -j REJECT
+		$ipt -A SS_SPEC_TPROXY -p udp --dport 80 -j DROP
+		$ipt -A SS_SPEC_TPROXY -p udp --dport 443 -j DROP
 		$ipt -A SS_SPEC_TPROXY -p udp $PROXY_PORTS -m set --match-set gfwlist dst -j TPROXY --on-port "$LOCAL_PORT" --tproxy-mark 0x01/0x01
 		$ipt -A SS_SPEC_TPROXY -p udp -m set --match-set gmlan src -m set ! --match-set china dst -j TPROXY --on-port "$LOCAL_PORT" --tproxy-mark 0x01/0x01
 		;;
