diff -ru a/package/openwrt-passwall2/luci-app-passwall2/root/usr/share/passwall2/app.sh b/package/openwrt-passwall2/luci-app-passwall2/root/usr/share/passwall2/app.sh
--- a/package/openwrt-passwall2/luci-app-passwall2/root/usr/share/passwall2/app.sh	2023-09-18 15:42:00.549863387 +0800
+++ b/package/openwrt-passwall2/luci-app-passwall2/root/usr/share/passwall2/app.sh	2023-09-18 15:42:47.426526797 +0800
@@ -205,7 +205,7 @@
 check_depends() {
 	local tables=${1}
 	if [ "$tables" == "iptables" ]; then
-		for depends in "iptables-mod-tproxy" "iptables-mod-socket" "iptables-mod-iprange" "iptables-mod-conntrack-extra" "kmod-ipt-nat"; do
+		for depends in "iptables-mod-tproxy" "iptables-mod-iprange" "iptables-mod-conntrack-extra" "kmod-ipt-nat"; do
 			[ -z "$(opkg status ${depends} 2>/dev/null | grep 'Status' | awk -F ': ' '{print $2}' 2>/dev/null)" ] && echolog "$tables透明代理基础依赖 $depends 未安装..."
 		done
 	else
